error.go

```diff
@@ -44,21 +44,21 @@ type HTTPError = jsonrpc.HTTPError
 
 // Client errors.
 const (
-	ErrBadCredentials strErr = "bad credentials"
-	ErrBadResponse    strErr = "bad response"
-	ErrBadRequest     strErr = "bad request"
+	ErrBadCredentials strError = "bad credentials"
+	ErrBadResponse    strError = "bad response"
+	ErrBadRequest     strError = "bad request"
 )
 
-type strErr string
+type strError string
 
-func (err strErr) Error() string { return string(err) }
+func (err strError) Error() string { return string(err) }
 
-// PathErrors describes issues for fields in a data-structure. Nested field
+// PathError describes issues for fields in a data-structure. Nested field
 // errors are reported with as `<parentField>.<subField>`. Field names are
 // camelCased to match the preferred case used for JSON encoding.
-type PathErrors map[string][]string
+type PathError map[string][]string
```

Comment:
I suggest we rename this back; this is a collection type and the name should remain plural.

For an error called `PathError`, I would expect a single error associated with a single path, and that's not what this is. If this is a linter suggestion, we should disable that linter.